### PR TITLE
Add `simplecov` to track test coverage

### PR DESCRIPTION
**Pull request by JamieMagee**
This is an initial addition of `simplecov` to Dependabot. Currently no level of coverage is enforced, but a report is generated.

From here, we can at least track coverage, and decide how and at what level we want to enforce code coverage.

Related to #9203

### CODE REVIEW

**Comment by jurre** [comment]
I think currently this will dump a report to stdout for each test run, but since the testsuite is split up heavily it will be hard to get an overall sense of where we stand? It would be nice to think about a way we can track coverage across the entire codebase

**Comment by JamieMagee** [comment]
It looks like we'd need a final step after all the unit test runs to collate the results. There are [some instructions in the simplecov documentation][1], but I'd need to investigate more on this.

[1]: https://github.com/simplecov-ruby/simplecov#merging-test-runs-under-different-execution-environments

**Comment by yeikel** [comment]
> It looks like we'd need a final step after all the unit test runs to collate the results. There are [some instructions in the simplecov documentation][1], but I'd need to investigate more on this.
> 
> [1]: https://github.com/simplecov-ruby/simplecov#merging-test-runs-under-different-execution-environments

Relevant article that might help 

https://rubyyagi.com/merge-simplecov-circleci/

File: common/spec/spec_helper.rb [hunk 1]
```diff
@@ -1,14 +1,32 @@
...
 
+# SimpleCov _must_ be started before any dependabot code is loaded
+SimpleCov.start do
+  add_filter "/spec/"
+  if ENV["CI"]
+    formatter SimpleCov::Formatter::SimpleFormatter
+  else
+    formatter SimpleCov::Formatter::MultiFormatter.new([
+      SimpleCov::Formatter::SimpleFormatter,
+      SimpleCov::Formatter::HTMLFormatter
```
Comment (yeikel) [inline common/spec/spec_helper.rb:22]: I understand the sentiment here, but I think that we should still generate the HTML report during the CI build and consider adding them as a build artifact for easy access

**Comment by yeikel** [comment]
Also, I was wondering if we could see if there is any integration we could add to include this as a pull request comment with a link to the HTML report as that will be helpful to raise awareness. Can we leverage Github pages for this?

**Comment by JamieMagee** [comment]
> Also, I was wondering if we could see if there is any integration we could add to include this as a pull request comment with a link to the HTML report as that will be helpful to raise awareness. Can we leverage Github pages for this?

I'm not sure about GitHub pages, but I have seen the [`joshmfrankel/simplecov-check-action`][1] and [Codecov supports simplecov] as well. I'll need to look more into it.

[1]: https://github.com/joshmfrankel/simplecov-check-action
[2]: https://about.codecov.io/tool/simplecov/

File: common/spec/spec_helper.rb (hunk 1 above)
Comment (JamieMagee) [reply common/spec/spec_helper.rb:22]: Fair point. I'll change that and use `actions/updload-artifact`.

Comment (JamieMagee) [reply common/spec/spec_helper.rb:22]: @yeikel looks like this was merged. I'll leave #9203 open and handle it in a follow-up PR.

**Pull request merged**